pauthabielf64/pauthabielf64.rst

```diff
@@ -861,198 +740,313 @@ main can. The static linker must communicate the details of how to
-  executable sections have Return Address Signing enabled.
+Every relocatable object, executable and shared library that uses the
+PAuth ABI ELF extensions must have a section named
+``.note.AARCH64-PAUTH-ABI-tag`` of type ``SHT_NOTE``. This section is
+structured as a note section as documented in SCO-ELF_.
 
-* AAELF64_ defines a dynamic tag ``DT_AARCH64_PAC_PLT`` that a static
-  linker must produce if the PLT sequences expect the .plt.got entries
-  to be signed by the dynamic linker.
+The name field (``namesz`` / ``name``) contains the string "ARM". The
+type field shall be 1. The ``descsz`` field must be at least 16, with
+the first 16 bytes of the description containing 2 64-bit words. With
+the first 64-bit word a platform identifier, and the second 64-bit
+identifier a version number for the ABI. When ``descsz`` is larger
+than 16 the remainder of the contents of desc are defined by the
+(platform id, version number).
 
-The dynamic tag can be reused assuming the PLT GOT is signed using the
-PACRET signing schema. The property and feature cannot be reused as
-the property is defined as: A set of processor features with which an
-ELF object or executable image is compatible, but does not require in
-order to execute correctly
+This ABI does not determine the format of the platform identifier. Arm
+reserves the platform id 0 for bare-metal no assoiciated
+platform. This also represents the empty string.
+
+The (platform id, version number) of (0, 0) is reserved as an invalid
+combination. The program cannot be run when pointer authentication is
+enabled.
 
-Proposal
---------
+The version id in ``.note.AARCH64-PAUTH-ABI-tag`` is not directly
+related to the version number of this document. It is controlled by
+the object-producer based on the signing schema that have been used
+for pointers.
+
+Base Compatibility Model
+------------------------
+
+A per ELF file marking scheme permits a coarse way of reasoning about compatibility.
+
+* The absence of a ``.note.AARCH64-PAUTH-ABI-tag`` section means no
+  information on how pointers are signed is available for this ELF
+  file.
+
+* The presence of a ``.note.AARCH64-PAUTH-ABI-tag`` means that the
+  pointers were signed in a compatible way with the default signing
+  rules for tuple (platform id, version number).
+
+* The static linker may fault the combination of relocatable
+  objects that contain ``.note.AARCH64-PAUTH-ABI-tag`` sections with
+  incompatible (platform id, version number) tuples. If an ELF file is
+  produced the output ``.note.AARCH64-PAUTH-ABI-tag`` must have the
+  invalid (platform id, version number) of (0, 0).
+
+* The combination of relocatable objects with
+  ``.note.AARCH64-PAUTH-ABI-tag`` and relocatable objects without a
+  ``.note.AARCH64-PAUTH-ABI-tag`` is not defined by this ABI.
+
+* A dynamic loader that encounters a (platform id, version number)
+  that it does not recognize, or the invalid (0, 0) then it must
```

Comment:
sentence is derailing. perhaps: .. that either encounters ... or the invalid (0, 0) marker, it must..

Reply:
I've made an attempt to reword.